elasticsearch-transport/lib/elasticsearch/transport/transport/http/manticore/manticore_sniffer.rb

```diff
@@ -0,0 +1,67 @@
+module Elasticsearch
+  module Transport
+    module Transport
+      module HTTP
+        class Manticore
+          # Handles node discovery ("sniffing")
+          #
+          class ManticoreSniffer < ::Elasticsearch::Transport::Transport::Sniffer
+            ES1_RE_URL  = /\[([^\/]*)?\/?([^:]*):([0-9]+)\]/
+            ES2_RE_URL  = /([^\/]*)?\/?([^:]*):([0-9]+)/
+
+            attr_reader   :transport
+            attr_accessor :timeout
+
+            # @param transport [Object] A transport instance
+            #
+            def initialize(*args)
+              @timeout = 1
+              @state_mutex = Mutex.new
+              @stopping = false
+              super
+            end
+
+            def sniff_every(interval_seconds, &block)
+              logger.info("Will sniff every #{interval_seconds} seconds for new Elasticsearch hosts!")
+              @state_mutex.synchronize do
+                return if @sniffer_thread
+                @sniffer_thread = Thread.new do
+                  last_sniff = Time.now
+                  until @state_mutex.synchronize { @stopping }
```

Comment:
I think that's just good form. Instance vars are `volatile` in JRuby now, but @headius mentioned that may change. AFAIK there's no guarantee that `@stopping` would ever have to become visible in another thread without some sort of synchronization. Now, that's almost impossible, but the point remains that we should `synchronize` on every cross-thread variable, if for no other reason than maintaining good discipline.